### PR TITLE
Test block insertion after title post blur

### LINKED ISSUE #3668: [Integration test] Inserter logic - Breakage case - Fix incorrect block insertion point after blurring post title
This issue is being addressed in [this PR](https://github.com/WordPress/gutenberg/pull/32944)

**PR where the breakage was fixed:** https://github.com/WordPress/gutenberg/pull/32831

## Tasks
1. Identify the block/component that is affected.
1. Add a test file for covering the breakage in the block/component folder.
1. Run the test and verify that it passes.
1. Revert the changes made for the fix.
1. Run the test again and verify that it fails.
1. _(Bonus)_ Add more test cases to the block/component.
1. Celebrate 🥳 !

### Notes
- If a test already exists in the block/component folder, review if the test cases can be migrated to the integration test.
- Some functionality might be not possible to replicate in the integration tests. In that case, indicate the blockers in the issue.

### PR DESCRIPTION
**Pull request by dcalhoun**
## Description
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/3668. Relates to #32831. Adds tests asserting that block inserter inserts new blocks in the correct position after blurring the post title field.

## How has this been tested?
1. Verify all CI tasks succeed.

## Screenshots 
n/a

## Types of changes
Automated testing.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by dcalhoun** [comment]
I need to test accessibility interactions before this PR is ready for review. It is quite possible that the current changes in the PR have broken/worsened audible descriptions for accessibility.

File: packages/edit-post/src/components/visual-editor/test/index.native.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		} );
+
+		// Focus first block
+		fireEvent.press( getByA11yLabel( /Paragraph Block. Row 1/ ) );
+
+		// Focus title
+		fireEvent( getAllByA11yLabel( 'Post title. test' )[ 0 ], 'select' );
+
+		// Add new Heading block
+		fireEvent.press( getByA11yLabel( 'Add block' ) );
```
Comment (fluiddot) [inline packages/edit-post/src/components/visual-editor/test/index.native.js]: I think you might need to include the following code to make the `FlatList` of the inserter menu's blocklist render the items:
```
		const blockList = getByTestId( 'InserterUI-Blocks' );
		// onScroll event used to force the FlatList to render all items
		fireEvent.scroll( blockList, {
			nativeEvent: {
				contentOffset: { y: 0, x: 0 },
				contentSize: { width: 100, height: 100 },
				layoutMeasurement: { width: 100, height: 100 },
			},
		} );
```

Comment (dcalhoun) [reply packages/edit-post/src/components/visual-editor/test/index.native.js]: Yeah, it is working currently as the Heading block is generally one of the first block types displayed and therefore scrolling is unneeded. However, it is possible that one day that may not be the case.

Comment (fluiddot) [reply packages/edit-post/src/components/visual-editor/test/index.native.js]: Oh that's right, the `FlatList` renders the first items without triggering the scroll event. In that case, we can leave it this way 👍 .

**Comment by fluiddot** [comment]
I see that the PR has some conflicts, they're caused due to [this PR](https://github.com/WordPress/gutenberg/pull/32506) got merged. Probably just updating it with `trunk` would be enough but let me know if you encounter any conflicts.

**Comment by dcalhoun** [comment]
@fluiddot these tests are ready for re-review. Please feel free to merge or close this PR as you see fit.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! Thanks @dcalhoun for adding this helpful test 🙇 .

_(And sorry for the super late review 😅 )_

**Pull request merged**